torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -4167,15 +4167,13 @@ at::Tensor XLANativeFunctions::as_strided(
     const at::Tensor& self, at::IntArrayRef size, at::IntArrayRef stride,
     std::optional<int64_t> storage_offset) {
   TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
-  const auto& base = bridge::GetXlaTensor(self)->Base();
-  const auto& tensor = base.defined() ? base : self;
-  XLATensorPtr self_tensor = bridge::GetXlaTensor(tensor);
+  XLATensorPtr self_tensor = bridge::GetXlaTensor(self);
```

Comment:
Since we are explicitly not supporting larger tensor use-case, what do you think about raising an error earlier (at tracing time), instead of later (at lowering time)?

Reply:
ah, is there an error message/check I can borrow?

Reply:
Hmmm. I don't think so. I think you have to check yourself whether the storage needed for the size-stride-storageoffset specification fits the current storage (i.e. number of elements in the tensor)

Reply:
There's a `at::detail::computeStorageNbytes` function in PyTorch.

Reply:
ok let me do that in a follow up pr, I want to merge this pr to unblock the customer.